ember-resources/src/core/class-based/resource.ts

```diff
@@ -174,6 +174,27 @@ export class Resource<Args = unknown> {
    */
   declare [Invoke]: ResourceHelperLike<Args, this>[typeof Invoke];
 
+  /**
+   * For use in the body of a class, without reactive arguments.
+   *
+   * `from` is what allows resources to be used in JS, they hide the reactivity APIs
+   * from the consumer so that the surface API is smaller.
+   *
+   * ```js
+   * import { Resource } from 'ember-resources';
+   *
+   * class SomeResource extends Resource {}
+   *
+   * class MyClass {
+   *   data = SomeResource.from(this);
+   * }
+   * ```
+   */
+  static from<SomeResource extends Resource<any>>(
+    this: Constructor<SomeResource>,
+    context: unknown
```

Comment:
that overload makes the type tests fail.. ie, _when_ this is a thunk, then the types are not checked against the args from signature. This needs to stay intact.

Also for me it is late in the eve right now and I will postpone this

Reply:
My alternative solution wasn't working either:

```ts
  static from<SomeResource extends Resource<any>, CT = unknown>(
    this: Constructor<SomeResource>,
    contextOrThunk: CT extends Function ? AsThunk<ArgsFrom<SomeResource>> : unknown
  ): SomeResource;
```